text/0268-acceptance-testing-refactor.md

```diff
@@ -0,0 +1,278 @@
+the versions from `@ember/test-helpers` directly (which means apps that have
+already adopted will have very minimal changes to make).
+
+The specific DOM helpers to be added to the `@ember/test-helpers` module are:
+
+```
+/**
+  Clicks on the specified selector.
+*/
+export function click(selector: string | HTMLElement): Promise<void>;
+
+/**
+  Taps on the specified selector.
+*/
+export function tap(selector: string | HTMLElement): Promise<void>;
+
+/**
+  Triggers a keyboad event on the specified selector.
+*/
+export function triggerKeyEvent(
+  selector: string | HTMLElement,
+  eventType: 'keydown' | 'keypress' | 'keyup',
+  keyCode: string,
+  modifiers?: {
+    ctrlKey: false,
+    altKey: false,
+    shiftKey: false,
+    metaKey: false
+  }
+): Promise<void>;
+
+/**
+  Triggers an event on the specified selector.
+*/
+export function triggerEvent(
+  selector: string | HTMLElement,
+  eventType: string,
+  eventOptions: any
+): Promise<void>;
+
+/**
+  Fill in the specified selector's `value` property with the provided text.
+*/
+export function fillIn(selector: string | HTMLElement, text: string): Promise<void>;
+
+/**
+  Focus the specified selector.
+*/
+export function focus(selector: string | HTMLElement): Promise<void>;
+
+/**
+  Unfocus the specified selector.
+*/
+export function blur(selector: string | HTMLElement): Promise<void>;
+
+/**
+  Returns a promise which resolves when the provided callback returns a truthy value.
+*/
+export function waitUntil(() => boolean): Promise<void>;
+
+/**
+  Returns a promise which resolves when the provided selector (and count) becomes present.
+*/
+export function waitFor(selector: string, count?: number): Promise<void>;
+```
+
+### `setupAcceptanceTest`
+
+This function will:
+
+* invoke `ember-test-helper`s `setupContext` with the tests context (which does the following):
+  * create an owner object and set it on the test context (e.g. `this.owner`)
+  * setup `this.pauseTest` and `this.resumeTest` methods to allow easy pausing/resuming
+    of tests
+* add routing related helpers
+  * setup importable `visit` method to visit the given url
+  * setup importable `currentRouteName` method which returns the current route name
+  * setup importable `currentURL` method which returns the current URL
+* add DOM interaction helpers (heavily influenced by @cibernox's lovely addon [ember-native-dom-helpers](https://github.com/cibernox/ember-native-dom-helpers))
+  * setup a getter for `this.element` which returns the DOM element representing
+    the applications root element
+  * setup importable `click` helper method
+  * setup importable `tap` helper method
+  * setup importable `triggerKeyEvent` helper method
+  * setup importable `triggerEvent` helper method
+  * setup importable `fillIn` helper method
+  * setup importable `focus` helper method
+  * setup importable `blur` helper method
+  * setup importable `waitUntil` helper method
+  * setup importable `waitFor` helper method
```

Comment:
- `waitFor` doesn't exist in `ember-native-dom-helpers`. So you like to have it added prior to the migration?

Reply:
I see that you haven't included `find` and `findAll` intentionally, with the idea that that users can just "use the platform". However, specially `find` compared to `this.element.querySelector()` is much more terse. Same goes for `findAll` vs `this.element.querySelectorAll()`, with the extra annoyance that `querySelectorAll` returns a `NodeList` instance instead of an Array, which is cumbersome to use to say the list.

I'd consider adding `find` and `findAll`. I believe that people will end up using them anyway.